src/ansys/sound/core/psychoacoustics/spectral_centroid.py

```diff
@@ -0,0 +1,142 @@
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Computes the spectral centroid of the signal."""
+import warnings
+
+from ansys.dpf.core import Field, Operator
+import numpy as np
+from numpy import typing as npt
+
+from . import PsychoacousticsParent
+from .._pyansys_sound import PyAnsysSoundException, PyAnsysSoundWarning
+
+
+class SpectralCentroid(PsychoacousticsParent):
+    """Computes the spectral centroid of a signal."""
+
+    def __init__(self, signal: Field = None):
+        """Create a ``SpectralCentroid`` object.
+
+        Parameters
+        ----------
+        signal: Field
+            Signal in Pa to compute spectral centroid on as a DPF field.
+        """
+        super().__init__()
+        self.signal = signal
+        self.__operator = Operator("compute_spectral_centroid")
+
+    @property
+    def signal(self):
+        """Signal."""
+        return self.__signal  # pragma: no cover
+
+    @signal.setter
+    def signal(self, signal: Field):
+        """Set the signal.
+
+        Parameters
+        -------
+        signal: Field
+            Signal in Pa to compute spectral centroid on as a DPF field.
+        """
+        self.__signal = signal
+
+    @signal.getter
+    def signal(self) -> Field:
+        """Signal.
+
+        Returns
+        -------
+        Field
+            Signal in Pa as a DPF field.
+        """
+        return self.__signal
+
+    def process(self):
+        """Compute the spectral centroid."""
+        if self.signal is None:
+            raise PyAnsysSoundException(
+                "No signal found for spectral centroid computation. Use 'SpectralCentroid.signal'."
+            )
+
+        self.__operator.connect(0, self.signal)
+
+        # Run the operator
+        self.__operator.run()
+
+        # Stores output
+        self._output = self.__operator.get_output(0, "double")
+
+    def get_output(self) -> float:
+        """Get the spectral centroid.
+
+        Returns
+        -------
+        float
+            Spectral centroid in Hz.
+        """
+        if self._output == None:
+            warnings.warn(
+                PyAnsysSoundWarning(
+                    "Output is not processed yet. \
+                        Use the 'SpectralCentroid.process()' method."
+                )
+            )
+
+        return self._output
+
+    def get_output_as_nparray(self) -> npt.ArrayLike:
+        """Get the spectral centroid as a NumPy array.
+
+        Returns
+        -------
+        numpy.ndarray:
+            Array of spectral centroid values in Hz.
+        """
+        return np.array([self._output])
```

Comment:
@anshlachamb This does not test  (and warns if relevant) whether output has been processed yet (same remarke as Alex for get_spectral_centroid). I suggest to use get_output.
Also, for the sake of consistency, in other recent classes, we rather chose to return empty arrays in this not-processed-yet case, rather than an array of None, which is what you would return here. I would also suggest to do the same here

Reply:
done